src/windows8/PushPluginProxy.js

```diff
@@ -7,24 +7,44 @@ var createNotificationJSON = function (e) {
 
     switch (e.notificationType) {
         case pushNotifications.PushNotificationType.toast:
-        notificationPayload = e.toastNotification.content.getXml();
-        break;
+            notificationPayload = e.toastNotification.content;
+            break;
 
         case pushNotifications.PushNotificationType.tile:
-        notificationPayload = e.tileNotification.content.getXml();
-        break;
+            notificationPayload = e.tileNotification.content;
+            break;
 
         case pushNotifications.PushNotificationType.badge:
-        notificationPayload = e.badgeNotification.content.getXml();
-        break;
+            notificationPayload = e.badgeNotification.content;
+            result.message = '';
+            result.count = notificationPayload.getElementsByTagName("badge")[0].getAttribute("value");
+            break;
 
         case pushNotifications.PushNotificationType.raw:
-        notificationPayload = e.rawNotification.content;
-        break;
+            result.message = e.rawNotification.content;
+            break;
     }
-    result.message = "";
-    result.xmlContent = notificationPayload;
-    result.objectReference = e;
+    
+    if (e.notificationType === pushNotifications.PushNotificationType.toast || e.notificationType === pushNotifications.PushNotificationType.tile) {
```

Comment:
Not a big deal but consider moving this code up in `case` arm. It will be easier to read.